src/foundry/client/pixi/webgl/shaders/lighting/base-lighting.d.mts

```diff
@@ -0,0 +1,182 @@
+     */
+    static get COLORATION_TECHNIQUES(): ReturnType<(typeof AdaptiveLightingShader)["getShaderTechniques"]>;
+
+    /**
+     * The coloration technique illumination shader fragment
+     */
+    static get ILLUMINATION_TECHNIQUES(): ReturnType<(typeof AdaptiveLightingShader)["getShaderTechniques"]>;
+
+    /**
+     * The coloration technique background shader fragment
+     */
+    static get BACKGROUND_TECHNIQUES(): ReturnType<(typeof AdaptiveLightingShader)["getShaderTechniques"]>;
+
+    /**
+     * The adjustments made into fragment shaders
+     */
+    static get ADJUSTMENTS(): string;
+
+    /**
+     * Contrast adjustment
+     */
+    static CONTRAST: string;
+
+    /**
+     * Saturation adjustment
+     */
+    static SATURATION: string;
+
+    /**
+     * Exposure adjustment
+     */
+    static EXPOSURE: string;
+
+    /**
+     * Switch between an inner and outer color, by comparing distance from center to ratio
+     * Apply a strong gradient between the two areas if attenuation uniform is set to true
+     */
+    static SWITCH_COLOR: string;
+
+    /**
+     * Shadow adjustment
+     */
+    static SHADOW: string;
+
+    /**
+     * Transition between bright and dim colors, if requested
+     */
+    static TRANSITION: string;
+
+    /**
+     * Incorporate falloff if a attenuation uniform is requested
+     */
+    static FALLOFF: string;
+
+    /**
+     * Compute illumination uniforms
+     */
+    static COMPUTE_ILLUMINATION: string;
+
+    /**
+     * Initialize fragment with common properties
+     */
+    static FRAGMENT_BEGIN: string;
+
+    /**
+     * Shader final
+     */
+    static FRAGMENT_END: string;
+
+    /**
+     * A mapping of available coloration techniques
+     */
+    static SHADER_TECHNIQUES: AdaptiveLightingShader.ShaderTechniques;
+    /**
+     * Determine the correct penalty to apply for a given darkness level and luminosity
+     * @param darknessLevel - The current darkness level on [0,1]
+     * @param luminosity    - The light source luminosity on [-1,1]
+     * @returns 0
+     * @deprecated since v12, until v14
+     * @remarks AdaptiveLightingShader#getDarknessPenalty is deprecated without replacement.
+     * The darkness penalty is no longer applied on light and vision sources.
+     */
+    getDarknessPenalty(darknessLevel: number, luminosity: number): 0;
+  }
+
+  namespace AdaptiveLightingShader {
+    type ShaderTypes = "coloration" | "illumination" | "background" | string;
+
+    interface ShaderTechniques extends Record<string, ShaderTechnique> {
+      LEGACY: ShaderTechnique;
+      LUMINANCE: ShaderTechnique;
+      INTERNAL_HALO: ShaderTechnique;
+      EXTERNAL_HALO: ShaderTechnique;
+      COLOR_BURN: ShaderTechnique;
+      INTERNAL_BURN: ShaderTechnique;
+      EXTERNAL_BURN: ShaderTechnique;
+      LOW_ABSORPTION: ShaderTechnique;
+      HIGH_ABSORPTION: ShaderTechnique;
+      INVERT_ABSORPTION: ShaderTechnique;
+      NATURAL_LIGHT: ShaderTechnique;
+    }
```

Comment:
I'd personally recommend leaving off the `extends Record<string, ShaderTechnique>`

If you're subclassing this you can always add your own properties regularly and if you're monkeypatching your own property in then you can declaration merge into this interface. However the DX of accessing `AdaptiveLightingShader.SHADER_TECHNIQUES` is reduced because you could write `AdaptiveLightingShader.SHADER_TECHNIQUES.LUMINACE`. This is a typo of `AdaptiveLightingShader.SHADER_TECHNIQUES.LUMINANCE` and you would never notice until at runtime it comes back as undefined.

Here's the entire pattern I'd recommend (make sure you're up to date on v12/main):
```ts
// This is so that even after an invalid declaration merge we can still as gracefully as possible continue (you can omit this comment in your diff).
type ShaderTechniques = ConformRecord<AdaptiveLightingShader, ShaderTechnique>

class AdaptiveLightingShader {
    static SHADER_TECHNIQUES: ShaderTechniques;
}

namespace AdaptiveLightingShader {
    interface ShaderTechniques { ... }
}

// This is to emit a compilation error if it's invalid. However the user can just ignore this error. (you can leave this comment off in your diff)
type _ShaderTechniquesMustBeValid = MustConform<ObjectToInterface<ShaderTechniques>, Record<string, ShaderTechnique>>;
```